### ISSUE TITLE
Implicit timestamps in treeEditor panel do not register for timestamps before January 01, 2000

### ISSUE THREAD

**Issue by bmcminn** [josdejong/jsoneditor#847]
I just discovered the "new" implicit timestamp labels for integers in the `treeEditor` panel, which is great because that's exactly the data I'm testing in my current data model.

**Problem:**
The timestamp recognition seems to only apply for positive integers greater than `946684800000 => January 01, 2000 @ 00:00:000Z`.

```json
[
  [ 1000000000000, "2001-09-09T01:46:40.000Z" ],
  [ 950000000000 , "2000-02-08T08:53:20.000Z" ],
  [ 946684800000 , "nothing" ],
  [ 946684800001 , "2000-01-01T00:00:001Z" ]
]
```

Not the worst thing in the world, but kind of disappointing how limited this utility is as a result.

**Idea for possible solution:**
Obviously the UI can't assume all integers greater than epoch (0) is a timestamp, but maybe having a toggle in the `#treeEditor .jsoneditor-menu` panel to force time stamp assumptions would be beneficial while leaving the implicit timestamps for  `>Jan 01, 2000` alone?

**Comment by josdejong** [comment]
That an interesting point, thanks for bringing this up. 

In the commit a615f7ab91b81fc8bf724b301cd6a6fa2e818f29 I've made it possible to apply your own rules by defining a function for option `timestampTag` (not yet published). Does that address your issue?

I'm very curious though which rules you are going to apply for your use case. Should we change something in the build-in rules? (which indeed treat a value from Jan 1, 2000 as a timestamp)

**Comment by bmcminn** [comment]
@josdejong I'd be curious to check out the commit and see, but that sounds like a decent enough solution to me.

For my use case I store everything in my database as a timestamp for simplicity sake because I know they can be consumed anywhere with minimal effort since Macs are insufferable jerks who demand date strings use a `T` delimiter or else "it's not a valid date"...

But in general, the timestamps I need correlate to legacy business data and user data dates, which usually go back a lot further than epoch, but in general I think your solution for a custom timestamp handler would work great.

**Comment by josdejong** [comment]
Thanks for your feedback.

Thinking about it, I think we should go one step further with the `timestampTag` function. My first approach passes value as the argument, but I think you often can know whether some field is a timestamp or not from the context, like if the field name is `creationTime` or `lastUpdated` etc. Depends on your data model. So we can make the `timestampTag` something like this (similar to other callbacks like `onEditable`):

```js
const options = { 
  timestampTag: function ({ path, field, value }) {
    return field === 'lastUpdated'
  }
}
```

**Comment by bmcminn** [comment]
I like the idea, but could be a bit cumbersome to adapt it anytime you moved on to a different model or had to rewrite your schema.

Heuristics might work here, wherein you provide a series of potential field names to match against, which has diminishing returns on performance the more variations you provide, but it would be simple enough to initially assume `date|time|updated|created|deleted`, etc string partials might be valid assumptions to apply a timestamp to.

**Comment by josdejong** [comment]
That's a great idea, using the field names instead of a value larger than some number. I've implemented your idea in 39dfc411009f67fe93854466c2c7358d007b4027.

**Comment by josdejong** [comment]
I've tuned it a bit more, made it now a combination of checking whether the value is an integer after Jan 200, or the field name contains something related to date or time (see 51410c363f62e3a8ad5f0407a026b228d132784b). I think we're there now.

**Comment by josdejong** [comment]
I've published the extended `timestampTag` in `v7.3.0`.

**Comment by josdejong** [comment]
I've reverted the rules introduced two weeks ago in `v7.3.0` because they can give issues with values containing timestamps in seconds instead of milliseconds, see  #856. With the two options `timestampTag` and `timestampFormat` you can tune timestamps exactly to your needs.
